docs/ui/components/FAQSection/index.tsx

```diff
@@ -0,0 +1,55 @@
+import Head from 'next/head';
+import React, { Children, PropsWithChildren, useMemo } from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+
+/** When wrapped around `Collapsible` components, this element generates the static ld+json required for Google search indexing. */
+export function FAQSection({ name, children }: PropsWithChildren<{ name: string }>) {
+  const mainEntity = useMemo(() => {
+    const faqs: { question: string; answer: string }[] = [];
+
+    Children.forEach(children, child => {
+      if (
+        child &&
+        typeof child === 'object' &&
+        'props' in child &&
+        child?.props?.mdxType === 'Collapsible'
+      ) {
+        const content = child.props.children;
+        faqs.push({
+          question: child.props.summary,
+          answer: renderToStaticMarkup(content).replace(
+            /\/\.\.\/\.\.\/\.\.\/\.\.\/\.\.\/\.\.\/\.\.\//g,
```

Comment:
Is there a way to resolve the URL in a way that doesn't couple this component to the path of the page it's used on? (Also this regex can be replaced with a much more readable string literal, but the bigger issue is that this FAQSection component assumes the links are going to be nested a specific level.)